ksql-common/src/main/java/io/confluent/ksql/util/KsqlConfig.java

```diff
@@ -181,6 +181,31 @@ void defineCurrent(final ConfigDef configDef) {
     }
   }
 
+  private static final Collection<CompatibilityBreakingStreamsConfig>
+      COMPATIBILITY_BREAKING_STREAMS_CONFIGS = ImmutableList.of(
+          new CompatibilityBreakingStreamsConfig(
+              StreamsConfig.TOPOLOGY_OPTIMIZATION,
+              StreamsConfig.NO_OPTIMIZATION,
+              StreamsConfig.OPTIMIZE)
+  );
+
+  private static final class CompatibilityBreakingStreamsConfig {
+    final String name;
+    final Object defaultValueOld;
+    final Object defaultValueCurrent;
+
+    CompatibilityBreakingStreamsConfig(final String name, final Object defaultValueOld,
+        final Object defaultValueCurrent) {
+      this.name = name;
```

Comment:
nit: null check.

Also, you could add check to ensure `name` is a valid Streams config name!